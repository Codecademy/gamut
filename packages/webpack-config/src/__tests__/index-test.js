const { createConfig } = require('../index');

describe('createConfig', () => {
  it('merges custom config', () => {
    const testConfig = createConfig()
      .common({
        context: './',
      })
      .merge({
        entry: 'my-file.js',
      })
      .toConfig();

    expect(testConfig.entry).toEqual('my-file.js');
  });

  it('allows overriding minimizer', () => {
    const testConfig = createConfig()
      .common({
        env: 'production',
        context: './',
        minimizer: 'cool',
      })
      .merge({
        entry: 'my-file.js',
      })
      .toConfig();

    expect(testConfig.optimization.minimizer).toEqual('cool');
  });
});
