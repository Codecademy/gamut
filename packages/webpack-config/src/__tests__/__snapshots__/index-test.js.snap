// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createConfig merges loaders 1`] = `
Object {
  "context": "/Users/jake/Sites/Codecademy/npm/gamut/packages/webpack-config/src/__tests__",
  "devtool": "cheap-module-eval-source-map",
  "entry": "/Users/jake/Sites/Codecademy/npm/gamut/packages/webpack-config/src/__tests__/src/main.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      [Function],
      Object {
        "loader": "file-loader",
        "test": /\\\\\\.jpe\\?g\\$\\|\\\\\\.gif\\$\\|\\\\\\.png\\$\\|\\\\\\.svg\\$\\|\\\\\\.woff2\\?\\$\\|\\\\\\.ttf\\$\\|\\\\\\.eot\\|\\\\\\.wav\\$\\|\\\\\\.mp3\\$/,
      },
    ],
  },
  "output": Object {
    "devtoolFallbackModuleFilenameTemplate": "[resourcePath]?[hash]",
    "devtoolModuleFilenameTemplate": "[absolute-resource-path]",
    "filename": "[name].js",
    "path": "/Users/jake/Sites/Codecademy/npm/gamut/packages/webpack-config/src/__tests__/dist",
  },
  "plugins": Array [
    DefinePlugin {
      "definitions": Object {
        "__DEV__": true,
        "process.env.NODE_ENV": "\\"development\\"",
      },
    },
    ContextReplacementPlugin {
      "newContentRecursive": undefined,
      "newContentRegExp": /en/,
      "newContentResource": undefined,
      "resourceRegExp": /moment\\[\\\\/\\\\\\\\\\]locale\\$/,
    },
  ],
  "resolve": Object {
    "extensions": Array [
      ".webpack.js",
      ".web.js",
      ".js",
      ".scss",
      ".css",
      ".json",
    ],
    "modules": Array [
      "node_modules",
      "node_modules/@codecademy",
    ],
  },
}
`;

exports[`createConfig merges loaders without mergeLoader 1`] = `
Object {
  "context": "/Users/jake/Sites/Codecademy/npm/gamut/packages/webpack-config/src/__tests__",
  "devtool": "cheap-module-eval-source-map",
  "entry": "/Users/jake/Sites/Codecademy/npm/gamut/packages/webpack-config/src/__tests__/src/main.js",
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "loader": "babel-loader",
        "options": Object {
          "cacheDirectory": true,
          "ignore": Array [
            "node_modules",
          ],
        },
        "test": /\\\\\\.js\\?\\$/,
      },
      Object {
        "loader": "file-loader",
        "test": /\\\\\\.jpe\\?g\\$\\|\\\\\\.gif\\$\\|\\\\\\.png\\$\\|\\\\\\.svg\\$\\|\\\\\\.woff2\\?\\$\\|\\\\\\.ttf\\$\\|\\\\\\.eot\\|\\\\\\.wav\\$\\|\\\\\\.mp3\\$/,
      },
      Object {
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "babel-loader",
            "options": Object {
              "plugins": Array [
                "@babel/plugin-transform-runtime",
              ],
              "presets": Array [
                "codecademy",
                "codecademy",
              ],
            },
          },
        ],
      },
    ],
  },
  "output": Object {
    "devtoolFallbackModuleFilenameTemplate": "[resourcePath]?[hash]",
    "devtoolModuleFilenameTemplate": "[absolute-resource-path]",
    "filename": "[name].js",
    "path": "/Users/jake/Sites/Codecademy/npm/gamut/packages/webpack-config/src/__tests__/dist",
  },
  "plugins": Array [
    DefinePlugin {
      "definitions": Object {
        "__DEV__": true,
        "process.env.NODE_ENV": "\\"development\\"",
      },
    },
    ContextReplacementPlugin {
      "newContentRecursive": undefined,
      "newContentRegExp": /en/,
      "newContentResource": undefined,
      "resourceRegExp": /moment\\[\\\\/\\\\\\\\\\]locale\\$/,
    },
  ],
  "resolve": Object {
    "extensions": Array [
      ".webpack.js",
      ".web.js",
      ".js",
      ".scss",
      ".css",
      ".json",
    ],
    "modules": Array [
      "node_modules",
      "node_modules/@codecademy",
    ],
  },
}
`;
